python/plugins/processing/core/parameters.py

```diff
@@ -79,13 +80,25 @@ def setValue(self, obj):
         of parameter.
         """
         if obj is None:
-            if not self.optional:
+            if not self.optional and self.default is None:
                 return False
-            self.value = None
-            return True
+            elif self.default is None:
+                self.value = None
+                return True
+            else:
```

Comment:
Don't want to beat this to death, but yes, I expect that __init__ should set the default value. That way when other people  use parameters (like perhaps your server code) You wouldn't have to check if the value for required parameters is `None`.